src/lib/vocab/community/communityServices.ts

```diff
@@ -238,44 +239,37 @@ export const LeaveCommunityService: ServiceByName['LeaveCommunity'] = {
 				persona_id,
 				community_id,
 			);
-			// TODO why can't this be parallelized? bug in our code? or the driver? failed to reproduce in the driver.
-			// const [personaResult, communityResult] = await Promise.all([
-			// 	repos.persona.findById(persona_id),
-			// 	repos.community.findById(community_id),
-			// ]);
 
 			unwrap(await checkCommunityAccess(persona_id, community_id, repos));
+
 			if (actor !== persona_id) {
 				return {ok: false, status: 403, message: 'actor does not have permission'};
 			}
 
-			const persona = unwrap(
-				await repos.persona.findById<Pick<ActorPersona, 'type' | 'community_id'>>(persona_id, [
-					'type',
-					'community_id',
-				]),
+			unwrap(await checkRemovePersona(persona_id, community_id, repos));
+
+			unwrap(await repos.assignment.deleteByPersonaAndCommunity(persona_id, community_id));
+
+			unwrap(await cleanOrphanCommunities([params.community_id], repos));
```

Comment:
Same cleanup in the identical line in the kick function below, it's offscreen so I can't suggest

```suggestion
			unwrap(await cleanOrphanCommunities([community_id], repos));
```